backend/src/v2/compiler/argocompiler/testdata/pipeline_with_env.yaml

```diff
@@ -0,0 +1,298 @@
+apiVersion: argoproj.io/v1alpha1
```

Comment:
There's no change to the argo compiler, it dumps IR into metadata fields. So I'm not sure if this json to yaml compilation is worth adding a separate test set. And we probably don't want to grow this folder with every single feature unless there's some argo compiler specifics to test.

Reply:
This is a great point. I will remove this test case.